src/plugins/lightscript.js

```diff
@@ -134,47 +134,80 @@ pp.parseObjectComprehension = function(node) {
   return this.finishNode(node, "ObjectComprehension");
 };
 
-// c/p parseBlock
-
-pp.parseWhiteBlock = function (allowDirectives?, isExpression?) {
-  const node = this.startNode(), indentLevel = this.state.indentLevel;
-  let allowEmptyBody = false;
-
-  // must start with colon or arrow
-  if (isExpression) {
-    this.expect(tt.colon);
-    if (!this.isLineBreak()) return this.parseMaybeAssign();
-  } else if (this.eat(tt.colon)) {
-    if (!this.isLineBreak()) return this.parseStatement(false);
-  } else if (this.eat(tt.arrow)) {
-    allowEmptyBody = true;
-
-    if (!this.isLineBreak()) {
-      if (this.match(tt.braceL)) {
-        // restart node at brace start instead of arrow start
-        const node = this.startNode();
-        this.next();
-        this.parseBlockBody(node, allowDirectives, false, tt.braceR);
-        this.addExtra(node, "curly", true);
-        return this.finishNode(node, "BlockStatement");
-      } else if (this.state.type.startsExpr) {
-        return this.parseMaybeAssign();
-      }
-    }
-  } else {
-    this.unexpected(null, "Whitespace Block must start with a colon or arrow");
+pp.couldBeginStatement = function() {
+  // ExpressionStatement
+  if (this.state.type.startsExpr) return true;
+
+  // c/p from parseStatement
+  switch (this.state.type) {
+    // break and continue only allowed within loops; omitted here
+    case tt._debugger:
+    case tt._do:
+    case tt._for:
+    case tt._function:
+    case tt._class:
+    case tt._if:
+    case tt._return:
+    case tt._switch:
+    case tt._throw:
+    case tt._try:
+    case tt._let:
+    case tt._const:
+    case tt._var:
+    case tt._now:
+    case tt._while:
+    case tt._with:
+    case tt.braceL:
+    case tt.semi:
+    case tt._export:
+    case tt._import:
+    case tt.name:
+      return true;
   }
 
-  // never parse directives if curly braces aren't used (TODO: document)
-  this.parseBlockBody(node, false, false, indentLevel);
-  this.addExtra(node, "curly", false);
+  return false;
+};
+
+pp.finishWhiteBlock = function(node, allowEmptyBody) {
   if (!allowEmptyBody && !node.body.length) {
     this.unexpected(node.start, "Expected an Indent or Statement");
   }
 
+  this.addExtra(node, "curly", false);
   return this.finishNode(node, "BlockStatement");
 };
 
+pp.parseInlineWhiteBlock = function(node, isExprContext, allowEmptyBody) {
+  if (this.state.type.startsExpr) return this.parseMaybeAssign();
+  // oneline statement case
+  node.body = this.couldBeginStatement() ? [this.parseStatement(true, false, isExprContext)] : [];
```

Comment:
Yes, I did try that approach as you suggested. I left an extensive comment about it which I assume got eaten or something?

It causes several dozen JS fixtures to break. The most interesting breakages involve ambiguities between `SequenceExpression`s and function argument lists. The breakages are caused by `parseStatement` going down a deeper parsing pathway than `parseMaybeAssign`, along which comma-separated expressions get turned into `SequenceExpression`s instead of unwinding the recursive descent.